### PR TITLE
Rewritten Logistic component as Expression.

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Logistic component rewritten as Expression. Extended docstring.

I don't really see the point of the b parameter in this definition of the logistic function, but did not want to touch the definition of the function.

### Progress of the PR
- [X] Change implemented (can be split into several points),
- [X] update docstring (if appropriate),
- [X] tested model fitting on simulated function with added Gaussian noise,
- [X] ready for review.

### CODE REVIEW

File: hyperspy/_components/logistic.py [hunk 1]
```diff
@@ -17,87 +17,58 @@
...
+            b=b,
+            c=c,
+            origin=origin,
+            position="origin",
+            module=module,
+            autodoc=False,
+            **kwargs)
+
+        # Boundaries
+        self.a.bmin = None
```
Comment (ericpre) [inline hyperspy/_components/logistic.py]: Is there any reason to set explicitly `bmin`, `bmax` to `None` for the different parameters? `None` should be already the default?

File: hyperspy/_components/logistic.py [hunk 2]
```diff
@@ -17,87 +17,58 @@
...
 
-class Logistic(Component):
+class Logistic(Expression):
 
-    """Logistic function component
+    """Logistic function (sigmoid/s-shaped curve) component
 
-    f(x) = a/(1+b*exp(-c*(x-origin)))
+    .. math::
+
```
Comment (ericpre) [inline hyperspy/_components/logistic.py:28]: Use latex to display nicely the formula (see for example https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/_components/lorentzian.py#L30)?

**Review by ericpre — CHANGES_REQUESTED** [review]
A few comments.

**Comment by jlaehne** [comment]
Removed the unneccessary default parameter settings.

**Comment by jlaehne** [comment]
Thanks Eric, I implemented the proposed changes.

File: hyperspy/_components/logistic.py (hunk 1 above)
Comment (jlaehne) [reply hyperspy/_components/logistic.py]: No, so removed.

File: hyperspy/_components/logistic.py (hunk 2 above)
Comment (jlaehne) [reply hyperspy/_components/logistic.py:28]: Good idea. I also improved the docstring further for consistency.

**Pull request closed without merging**